tool/net/redbean.c

```diff
@@ -4697,6 +4697,162 @@ static int LuaGetHttpReason(lua_State *L) {
   return 1;
 }
 
+static int EncodeJsonData(lua_State *L, char **buf, int level, char *numformat) {
+  size_t idx = -1;
+  size_t tbllen, buflen;
+  bool isarray;
+  int t = lua_type(L, idx);
+  if (level < 0) return luaL_argerror(L, 1, "too many nested tables");
+  if (LUA_TSTRING == t) {
+    appendf(buf, "\"%s\"", gc(EscapeJsStringLiteral(lua_tostring(L, idx), -1, 0)));
```

Comment:
Optionally, if string performance is a concern, you could nudge out a bit more of it here by breaking it down into `appendw('"');appendd(...);appendw('"');` That way you avoid spinning up the `__fmt()` engine and avoid the strlen scan.

Reply:
Yes, I'll switch to that.